### PR TITLE
Create `@kampus/gql-utils` package

### PR DESCRIPTION
**Pull request by usirin**
A set of utilities to power our gql server. isolated to an npm workspace.

### CODE REVIEW

File: packages/gql-utils/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (Ketcap) [inline packages/gql-utils/.gitignore:1]: Empty file ?

**Review by Ketcap — APPROVED** [review]
LGTM!

File: packages/gql-utils/prisma/__mocks__/prisma.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { beforeEach } from "vitest";
+import { mockDeep, mockReset } from "vitest-mock-extended";
+
```
Comment (aecepoglu) [inline packages/gql-utils/prisma/__mocks__/prisma.ts:3]: You may find https://github.com/thoughtbot/fishery useful. It is just a utility layer that provides factorios for creating test data.

File: packages/gql-utils/prisma/create-loader.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+ * });
+ *
+ * const userByID = byID.load("some-id");
+ * const userByUsername = byUsername.load("testuser");
+ * ```
+ *
+ * @param model - prisma model
+ * @param identifier - the field that's the keys are gonna match against.
+ * @param onFetchComplete - an optional function that's gonna be called after prisma fetch is complete.
+ * @returns a new DataLoader instance that's ready to load data db using prisma model
```
Comment (aecepoglu) [inline packages/gql-utils/prisma/create-loader.ts:36]: "ready to load data _into_ db" maybe?

File: packages/gql-utils/prisma/create-loader.test.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        deletedAt: null,
+      },
+    });
+  });
+
+  it("returns an error when trying to load a model that doesn't exist", async () => {
+    mockedPrisma.user.findMany.mockResolvedValueOnce([mockUser({ id: "2" })]);
+
+    const byID = createPrismaLoader(mockedPrisma.user, "id");
+    await expect(() => byID.load("1")).rejects.toThrowError("not found: id: 1");
```
Comment (aecepoglu) [inline packages/gql-utils/prisma/create-loader.test.ts:30]: Do we want to define our errors and expose them so we can match against an `IdNotFoundError` vs a string?

**Review by aecepoglu — APPROVED** [review]

File: packages/gql-utils/prisma/create-loader.test.ts (hunk 4 above)
Comment (usirin) [reply packages/gql-utils/prisma/create-loader.test.ts:30]: That's a really great suggestion. We should create a ticket and get it worked on. @aecepoglu can you help with that?

**Pull request merged**